File: edupilot_api/EduPilot.Api/DTOs/StudentLoginRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace EduPilot.Api.Models
{
    public class StudentLoginRequestDTO
    {
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [DataType(DataType.Password)]
        public string Password { get; set; }
    }
}

File: edupilot/lib/models/dtos/student_login_request_dto.dart
class StudentLoginRequestDTO {
  final String email;
  final String password;

  const StudentLoginRequestDTO({
    required this.email,
    required this.password,
  });

  factory StudentLoginRequestDTO.fromJson(Map<String, dynamic> json) =>
      StudentLoginRequestDTO(
        email: json['email'] as String,
        password: json['password'] as String,
      );

  Map<String, dynamic> toJson() => {
        'email': email,
        'password': password,
      };

}